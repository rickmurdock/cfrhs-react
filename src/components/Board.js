import React from 'react';
import contacts from '../data/contacts';

const Board = () => {
  let boardMembers = contacts.filter(contact => contact.board === true);
  let boardMembers1 = boardMembers.slice(0,5);
  let boardMembers2 = boardMembers.slice(5);
  
  let boardColumn1 = boardMembers1.map((board1, index) => {  
    return (
      <div key={index}>
        <h5>{board1.position}</h5>
        <p>{board1.name}</p>
      </div>
    );
  });

  let boardColumn2 = boardMembers2.map((board2, index) => {  
    return (
      <div key={index}>
        <h5>{board2.position}</h5>
        <p>{board2.name}</p>
      </div>
    );
  });

  return (
    <div className="container-fluid">
      <h2 className="text-center page-title my-3">Board of Directors</h2>
      <div className="row my-3">
        <div className="col-md-3"></div>
        <div className="col-md-4">
          {boardColumn1}
        </div>
        <div className="col-md-3">
          {boardColumn2}
        </div>
        <div className="col-md-2"></div>
      </div>
    </div>
  );
};

export default Board;