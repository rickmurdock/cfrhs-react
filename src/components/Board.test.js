import React from 'react';
import renderer from 'react-test-renderer';
import Board from './Board';

test('Board renders correctly', () => {
    const tree = renderer.create(
        <Board />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});