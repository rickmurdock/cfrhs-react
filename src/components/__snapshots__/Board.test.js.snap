// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<div
  className="container-fluid"
>
  <h2
    className="text-center page-title my-3"
  >
    Board of Directors
  </h2>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    />
    <div
      className="col-md-4"
    >
      <div>
        <h5>
          President
        </h5>
        <p>
          Jerry Honetor
        </p>
      </div>
      <div>
        <h5>
          Vice President - Membership & Programs
        </h5>
        <p>
          Phil Piet
        </p>
      </div>
      <div>
        <h5>
          Vice President - Education & Public Relations
        </h5>
        <p>
          Irv Lipscomb
        </p>
      </div>
      <div>
        <h5>
          Secretary
        </h5>
        <p>
          Joe Lehmann, Jr.
        </p>
      </div>
      <div>
        <h5>
          Assistant Secretary
        </h5>
        <p>
          Jarrod Reynolds
        </p>
      </div>
    </div>
    <div
      className="col-md-3"
    >
      <div>
        <h5>
          Treasurer
        </h5>
        <p>
          David Rhea
        </p>
      </div>
      <div>
        <h5>
          Assistant Treasurer
        </h5>
        <p>
          Roger Wilson
        </p>
      </div>
      <div>
        <h5>
          Museum Curator
        </h5>
        <p>
          Ken Murdock
        </p>
      </div>
      <div>
        <h5>
          Historian
        </h5>
        <p>
          Phil Cross
        </p>
      </div>
      <div>
        <h5>
          Immediate Past President
        </h5>
        <p>
          Bill Dusenbury
        </p>
      </div>
    </div>
    <div
      className="col-md-2"
    />
  </div>
</div>
`;
